src/autoscaler/pruner/appmetricsdb.go

```diff
@@ -1,59 +1,30 @@
 package pruner
 
 import (
-	"time"
-
 	"autoscaler/db"
 
 	"code.cloudfoundry.org/clock"
 	"code.cloudfoundry.org/lager"
 )
 
 type AppMetricsDbPruner struct {
-	logger       lager.Logger
 	appMetricsDb db.AppMetricDB
-	interval     time.Duration
 	cutoffDays   int
 	clock        clock.Clock
-	doneChan     chan bool
+	logger       lager.Logger
 }
 
-func NewAppMetricsDbPruner(logger lager.Logger, appMetricsDb db.AppMetricDB, interval time.Duration, cutoffDays int, clock clock.Clock) *AppMetricsDbPruner {
+func NewAppMetricsDbPruner(appMetricsDb db.AppMetricDB, cutoffDays int, clock clock.Clock, logger lager.Logger) *AppMetricsDbPruner {
 	return &AppMetricsDbPruner{
-		logger:       logger,
+
 		appMetricsDb: appMetricsDb,
-		interval:     interval,
 		cutoffDays:   cutoffDays,
 		clock:        clock,
-		doneChan:     make(chan bool),
-	}
-}
-
-func (amdp *AppMetricsDbPruner) Start() {
-	go amdp.startMetricPrune()
-
-	amdp.logger.Info("app-metrics-db-pruner-started", lager.Data{"refresh_interval_in_hours": amdp.interval})
-}
-
-func (amdp *AppMetricsDbPruner) Stop() {
-	close(amdp.doneChan)
-	amdp.logger.Info("app-metrics-db-pruner-stopped")
-}
-
-func (amdp *AppMetricsDbPruner) startMetricPrune() {
-	ticker := amdp.clock.NewTicker(amdp.interval)
-	for {
-		amdp.PruneOldMetrics()
-		select {
-		case <-amdp.doneChan:
-			ticker.Stop()
-			return
-		case <-ticker.C():
-		}
+		logger:       logger,
 	}
 }
 
-func (amdp *AppMetricsDbPruner) PruneOldMetrics() {
+func (amdp AppMetricsDbPruner) PruneOldData() {
 	amdp.logger.Debug("Prune app metric db data older than", lager.Data{"cutoff_days": amdp.cutoffDays})
```

Comment:
Just state what you are doing, like pruning app metrics. The moment you add another piece of data, the debug statement won't make any sense.